src/eko/matchings.py

```diff
@@ -0,0 +1,201 @@
+r"""Holds the classes that define the |FNS|."""
+import logging
+from dataclasses import dataclass
+from typing import List, Union
+
+import numpy as np
+
+from .io.types import EvolutionPoint as EPoint
+from .io.types import FlavorIndex, FlavorsNumber, SquaredScale
+from .quantities.heavy_quarks import MatchingScales
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass(frozen=True)
+class Segment:
+    """Oriented path in the threshold landscape."""
+
+    origin: SquaredScale
+    """Starting point."""
+    target: SquaredScale
+    """Final point."""
+    nf: FlavorsNumber
+    """Number of active flavors."""
+
+    @property
+    def is_downward(self) -> bool:
+        """Return True if ``origin`` is bigger than ``target``."""
+        return self.origin > self.target
+
+    def __str__(self):
+        """Textual representation, mainly for logging purpose."""
+        return f"PathSegment({self.origin} -> {self.target}, nf={self.nf})"
+
+
+Path = List[Segment]
+
+
+@dataclass(frozen=True)
+class Matching:
+    """Matching between two different segments.
+
+    The meaning of the flavor index is the PID of the corresponding heavy quark.
```

Comment:
This refers to the index in the *upper* patch, correct? We need to have an absolute sorting and you need to mention that here ... (think of: we evolve from 20 -> 30 then we add the bottom matching instead if we evolve 30 -> 20 we have to remove the bottom matching)

Reply:
As mentioned in person: `hq` is a `FlavorIndex`, not a `FlavorsNumber`, so it refers to the PID of the heavy quark associated to the matching, e.g. 5 for bottom.

It is not the lower, nor the upper patch. Of course, numerically it coincides with the `FlavorsNumber` in the upper patch, but this is a consequence, conceptually they are different objects.